### PR TITLE
Derivs: Use GF3D2 as input in calc_derivs

### PR DESCRIPTION
**Pull request by lwJi**

### CODE REVIEW

File: Derivs/src/derivs.cxx [hunk 1]
```diff
@@ -10,9 +10,29 @@ using namespace Loop;
 
 template <int CI, int CJ, int CK, typename T>
 CCTK_ATTRIBUTE_NOINLINE void
-calc_derivs(const vec<GF3D5<T>, dim> &dgf, const GridDescBaseDevice &grid,
-            const GF3D5<const T> &gf, const GF3D5layout layout,
-            const vect<T, dim> dx, const int deriv_order) {
+calc_copy(const GF3D5<T> &gf0, const GF3D5layout layout,
```
Comment (eschnett) [inline Derivs/src/derivs.cxx:13]: The output variables have the suffix `0`, the input variables the suffix `1`. That's a bit weird. Can you use `0` for the input variables, and use no suffix for the output variables?

File: Derivs/src/derivs.hxx [hunk 2]
```diff
@@ -332,19 +332,18 @@ inline CCTK_ATTRIBUTE_ALWAYS_INLINE
...
   using namespace Loop;
   // We use explicit index calculations to avoid unnecessary integer
   // multiplications
-  const T *restrict const ptr = &gf(layout, I);
+  const T *restrict const ptr = &gf(I);
   const std::array<std::ptrdiff_t, Loop::dim> offsets{
-      layout.delta(1, 0, 0),
-      layout.delta(0, 1, 0),
-      layout.delta(0, 0, 1),
+      gf.delta(1, 0, 0),
```
Comment (eschnett) [inline Derivs/src/derivs.hxx:344]: I don't think these lines are correct. I don't think that the grid function class `GF3D2` has a `delta` member function.

**Review by eschnett — CHANGES_REQUESTED** [review]
I added comments requesting changes.

File: Derivs/src/derivs.cxx (hunk 1 above)
Comment (lwJi) [reply Derivs/src/derivs.cxx]: I modify suffix as suggested.

**Comment by lwJi** [comment]
Hi @eschnett , I have addressed your comments. please take a look again. Thanks

File: Derivs/src/derivs.hxx (hunk 2 above)
Comment (eschnett) [reply Derivs/src/derivs.hxx:344]: nice!

**Review by eschnett — APPROVED** [review]

**Pull request merged**